### PR TITLE
Add more runtime logging

### PR DESCRIPTION
**Pull request by Bensk1**
The reviews requested more information on the number of created what if indexes and the time spent in cost estimation and index creation. This PR should integrate it properly.

It also contains a tiny bit of refactoring which only slightly improves the what_if module. I think there is an issue to refactor it.

### CODE REVIEW

File: selection/database_connector.py [hunk 1]
```diff
@@ -17,7 +24,7 @@ def exec_fetch(self, statement, one=True):
         return self._cursor.fetchall()
 
     def enable_simulation(self):
-        pass
+        raise NotImplementedError
```
Comment (Bensk1) [inline selection/database_connector.py:27]: Refactoring, nothing to do with PR.

File: selection/what_if_index_creation.py [hunk 2]
```diff
@@ -22,12 +22,8 @@ def simulate_index(self, potential_index, store_size=False):
 
     def drop_simulated_index(self, index):
         oid = index.hypopg_oid
+        self.db_connector.drop_simulated_index(oid)
         del self.simulated_indexes[oid]
-        self._drop_index_by_id(oid)
-
-    def _drop_index_by_id(self, oid):
-        statement = f"select * from hypopg_drop_index({oid})"
-        self.db_connector.exec_only(statement)
```
Comment (Bensk1) [inline selection/what_if_index_creation.py:30]: Refactoring, nothing to do with PR.

File: selection/database_connector.py [hunk 3]
```diff
@@ -7,6 +8,12 @@ def __init__(self, db_name, autocommit=False):
         self.autocommit = autocommit
         logging.debug("Database connector created: {}".format(db_name))
 
+        # This does not reflect the number of unique simulated indexes but
+        # the number of simulate_index calls
+        self.simulated_indexes = 0
+        self.cost_estimation_duration = 0
```
Comment (klauck) [inline selection/database_connector.py:14]: I would prefer storing also the number of cost estimations here, although they can be calculated by `CostEstimation.cost_requests - CostEstimation.cache_hits`.

File: selection/database_connector.py [hunk 4]
```diff
@@ -43,6 +50,30 @@ def _prepare_query(self, query):
             elif "select" in query_statement or "SELECT" in query_statement:
                 return query_statement
 
+    def simulate_index(self, index):
+        self.simulated_indexes += 1
+
+        start_time = time.time()
+        result = self._simulate_index(index)
+        end_time = time.time()
+        self.index_simulation_duration += end_time - start_time
```
Comment (klauck) [inline selection/database_connector.py:59]: I usually use and would prefer `self.index_simulation_duration += time.time() - start_time`.

File: selection/database_connector.py [hunk 5]
```diff
@@ -43,6 +50,30 @@ def _prepare_query(self, query):
...
+
+        return result
+
+    def drop_simulated_index(self, identifier):
+        start_time = time.time()
+        self._drop_simulated_index(identifier)
+        end_time = time.time()
+        self.index_simulation_duration += end_time - start_time
```
Comment (klauck) [inline selection/database_connector.py:67]: see above

File: selection/database_connector.py [hunk 6]
```diff
@@ -43,6 +50,30 @@ def _prepare_query(self, query):
...
+
+    def get_cost(self, query):
+        start_time = time.time()
+        cost = self._get_cost(query)
+        end_time = time.time()
+        self.cost_estimation_duration += end_time - start_time
```
Comment (klauck) [inline selection/database_connector.py:73]: see above

**Comment by Bensk1** [comment]
Thanks! I included one of your proposals.

**Pull request merged**